Integration testing index creation. DSDEGP-1340

kshakir opened the pull request:
### Description

The Python integration tests now contact Elasticsearch to confirm that clio created the index correctly.
DSDEGP-1340

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: Ensure that you have added or updated tests
- [x] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509).
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Move the JIRA issue to QA once this checklist is completed

pshapiro4broad commented on src/test/resources/cliointegrationtest/clio_test.py:
```diff
@@ -48,8 +54,51 @@ def test_health():
+    mappings = js['readgroups']['mappings']['default']
+    assert mappings['dynamic'] == 'false'
+    fields = mappings['properties']
+    assert fields['analysis_type']['type'] == 'keyword'
```
Could this be made data driven, e.g. using a structure that maps 'keyword' to all the keyword fields 'date' to the date fields, etc. Then all these asserts could be replaced with a loop (or nested loops) and a single assert

pshapiro4broad commented on src/test/resources/cliointegrationtest/clio_test.py:
```diff
@@ -11,6 +11,12 @@
+elasticsearch_http_scheme = os.environ.get('ELASTICSEARCH_HTTP_SCHEME', 'http')
+elasticsearch_http_host = os.environ.get('ELASTICSEARCH_HTTP_HOST', 'localhost')
+elasticsearch_http_port = os.environ.get('ELASTICSEARCH_HTTP_PORT', 9200)
+elasticsearch_http_uri = '{0}://{1}:{2}'.format(
```
Maybe make a method or string constant to avoid the duplication when creating a URI from scheme/host/port?

pshapiro4broad reviewed (approved):
:+1: minor comments

kshakir commented:
@pshapiro4broad Take another gander and let me know if you have comments. Caveats that all of this Python may disappear if we like the direction that firecloud is going by moving to scalatest for integration testing.

pshapiro4broad commented on src/test/resources/cliointegrationtest/clio_test.py:
```diff
@@ -62,10 +71,86 @@ def test_bad_path():
+    field_names_by_type = __get_field_names_by_type(mapping_properties)
+    __assert_expected_fields(
+        field_names_by_type,
+        {
```
Thanks, I find this much easier to read. We can keep the basic structure if we decide to migrate to scala.

kshakir committed:
Integration testing index creation.
Elasticsearch host passed to python script.
Removed separated subnets that were only allowing the python to talk to clio, and not ES directly.
More debugging info when the python tests go wrong.
